salt/states/file.py

```diff
@@ -352,6 +352,33 @@ def _check_user(user, group):
     return err
 
 
+def _is_valid_relpath(
+        relpath,
+        maxdepth=None):
+    '''
+    Performs basic sanity checks on a relative path.
+
+    Requires slash-separated path components (i.e. the kind obtained
+    through cp.list_master or other such calls).
+
+    Ensures that the path does not contain directory transversal, and
+    that it does not exceed a stated maximum depth (if specified).
+    '''
+    # Check relpath surrounded by slashes, so that `..` can be caught as
+    # a path component at the start, end, and in the middle of the path.
+    if '/../' in '/{0}/'.format(relpath):
+        return False
+
+    # Check that the relative path's depth does not exceed maxdepth
+    if maxdepth is not None:
+        # Since paths are all master, just use POSIX separator
+        path_depth = relpath.strip('/').count('/')
```

Comment:
You can't assume this. We're working on getting the master running on Windows, so you must use ``os.sep`` here.